src/services/postService.js

```diff
@@ -1,15 +1,36 @@
 const uuid = require("uuid");
 const { throwIfError, CLASS_POST } = require('../utilities/dynamoUtilities');
 const postDAO = require("../repository/postDAO");
+const { getSongs } = require("./songService");
 
-const createPost = async (userId, description, score, title, tags) => {
+const createPost = async (userId, description, score, title, artist, tags) => {
     let tagMap = new Map();
     if (tags){
         for (const i of tags){
             tagMap.set(i, true);
         }
     }
-    const post = { class: CLASS_POST, itemID: uuid.v4(), postedBy: userId, description, score, title, replies: [], likedBy: [], tags: tagMap, isFlagged: 0 };
+    const result = await getSongs({track: title, artist}, 0);
```

Comment:
I would make this song request stuff all frontend work. Allow the user to search for a song. They select one, then you take the title and artist and append it to the post body. It's be 2 different backend calls. One for searching songs which are then displayed to the user so they can select one. Then once it is selected, take the data and make a call to create a post with that data. That way there's algorithm to make a best guess at what the user wants. We just simply pass the data in to create a post